### DISCUSSION TITLE
SSE Connection to Claude Desktop: Connecting Versus Spinning up.

### DISCUSSION THREAD

**Discussion by andrewlayer** [modelcontextprotocol/.github#91] [Q&A] [answered]
### Pre-submission Checklist

- [X] I have verified that this discussion would not be more appropriate as an issue in a specific repository
- [X] I have searched existing discussions to avoid duplicates

So I have the following [repository](https://github.com/andrewlayer/mcptest/blob/main/run.py), I have an http server that I spin up separately using python `run.py`.  This server is hosted at `http://localhost:8000/sse`.  I am able to successfully connect to this server using the inspector application and I can get resources, tools, prompts, etc.  I can't seem to connect to the claude desktop client because I can't seem to figure out how to configure the client to use SSE instead of STDIO.

I understand that the MCP team is working on "remote MCP servers".  I am not entirely sure how these would differ from the sample application I have running and connecting to the inspector application.  What would benefit would a remote MCP server have over connecting via SSE?

My last question it, why are each of the clients responsible for spinning up the MCP servers.  From my understanding this would mean that each client will attempt to spin up an instance of an MCP server.  So if I have 2 MCP clients active attempting to access the same server, two instances of that server will be spun up (or more likely, one will fail).  I almost feel like spinning up servers as docker containers that you connect to locally would be ideal because that way you only ever have a single running instance of a server.

Can't appreciate enough your team putting together such a high quality protocol.  I have been yearning for something similar to this for quite some time.

**Comment by jspahrsummers** [comment, marked as answer]
Claude Desktop only connects to servers over stdio right now, not SSE. When we refer to the chunk of work involved in "remote servers," we mean generally enhancing everything _not_ stdio. See https://github.com/modelcontextprotocol/specification/discussions/112 for a collection of some of the work related to that.

Clients only start servers when using the stdio transport, as this method of communication involves a parent–child relationship. Servers accessible over a socket won't have this same behavior, and could accept multiple connections on a single instance.
